ruby-on-rails/docs/gem-tutorials/spree.md

```diff
@@ -0,0 +1,324 @@
+
+```ruby
+class AddDataToUsers < ActiveRecord::Migration[5.1]
+  def change
+    add_column :spree_users, :gender, :string, default: 'other'
+    add_column :spree_users, :first_name, :string
+    add_column :spree_users, :last_name, :string
+    add_column :spree_users, :birthday, :date, default: '2000-01-01'
+  end
+end
+```
+
+The table is in spree module so even if the model name is *User*, **spree_** is required before model name. The rest is define as usual. Version of migration (here 5.1) need to be specify.
+
+##### Model modification
+
+Now, we are going to add some adaptation to the user model from spree. Spree official documentation advice to use decorator to modify his own class. So here we will create a file user_decorator.rb and modify the class with **.class_eval**.
+
+
+```ruby
+#/app/models/spree/user_decorator.rb
+Spree::User.class_eval do
+  validates :login, presence: true, uniqueness: true
+
+  validates :birthday, timeliness: { on_or_before: -> { Date.current }, type: :date }
+  # Here we do not use an integer with enum to avoid the error:
+  # You tried to define an enum named "gender" on the model "Spree::User",
+  # but this will generate a instance method "female?", which is already defined by another enum.
+  validates :gender, inclusion: %w[female male other]
+
+  has_many :prescriptions, class_name: 'Spree::Prescription', dependent: :destroy
+end
+```
+
+Here nothing strange, we define validates and relation as usual. We will show how define **belongs_to** on the next example. It’s also possible to define method for the model. In need to modificate an existing method from spree gem, it is sufficient to define here a method with the same name and write custom code on it.
+
+
+### Generate a new model
+Here we are going to add a new model prescription which belongs to an user.
+
+##### Migration
+
+In this migration a nex model Prescription is create, but we need to specify spree_ to specify the appartenance to spree module. The definition of the fields is usual. To generate the migration do as usual with rails generate model...
+
+```ruby
+class CreateSpreePrescriptions < ActiveRecord::Migration[5.1]
+  def change
+    create_table :spree_prescriptions do |t|
+      t.decimal :add_right, default: 0.0, null: false
+      t.decimal :add_left, default: 0.0, null: false
+      t.boolean :pupillary_distance_double
+      t.string :title, default: '', null: false
+
+      t.belongs_to :user                                                                                                          
+      t.timestamps
+    end
+  end
+end
+```
+
+The main difference as usual migration, is the **belongs_to**. Normally we use **references**, but with the spree module to avoid some errors, is advised to use **belongs_to**.
```

Comment:
The main difference with an usual migration

But with the spree module, to avoid some errors